scripts/release.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env bash
+#
+# Copyright Kroxylicious Authors.
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+
+set -e
+REPOSITORY="origin"
+BRANCH_FROM="main"
+DRY_RUN="false"
+SKIP_VALIDATION="false"
+while getopts ":a:f:v:b:r:k:ds" opt; do
+  case $opt in
+    v) RELEASE_VERSION="${OPTARG}"
+    ;;
+    b) BRANCH_FROM="${OPTARG}"
+    ;;
+    r) REPOSITORY="${OPTARG}"
+    ;;
+    k) GPG_KEY="${OPTARG}"
+      if [[ -z "${GPG_KEY}" ]]; then
+          echo "GPG_KEY not set unable to sign the release. Please specify -k <YOUR_GPG_KEY>" 1>&2
+          exit 1
+      fi
+    ;;
+    d) DRY_RUN="true"
+    ;;
+    s) SKIP_VALIDATION="true"
+    ;;
+    \?) echo "Invalid option -${OPTARG}" >&2
+    exit 1
+    ;;
+  esac
+
+  case ${OPTARG} in
+    -*) echo "Option $opt needs a valid argument"
+    exit 1
+    ;;
+  esac
+done
+
+GREEN='\033[0;32m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+ORIGINAL_WORKING_BRANCH=$(git branch --show-current)
+
+cleanup() {
+    if [[ -n ${ORIGINAL_WORKING_BRANCH} ]]; then
+        git checkout "${ORIGINAL_WORKING_BRANCH}" || true
+    fi
+
+    if [[ ${TEMPORARY_RELEASE_BRANCH} ]]; then
+        # Note that git branch -D echos the sha of the deleted branch to
+        # stdout.  This is great for debugging the release process as it
+        # lets the developer restore to the state of the tree.
+        git branch -D "${TEMPORARY_RELEASE_BRANCH}" || true
```

Comment:
You'll see:
```
Deleted branch prepare-release-2023-10-22 (was 1af9612b).
Deleted tag 'v0.3.0' (was 1af9612b)
```

so you can `git checkout 1af9612b` to get back to the state created by the release script.